x-pack/plugins/apm/server/lib/settings/apm_indices/index.ts

```diff
@@ -0,0 +1,75 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { SavedObjectsClient } from 'kibana/server';
+import { ApmIndicesConfig, getApmIndices } from './get_apm_indices';
+import { APMConfig } from '../../../index';
+
+type ISavedObjectsClient = Pick<SavedObjectsClient, 'get'>;
+
+interface ApmIndicesSettings {
+  indices?: ApmIndicesConfig;
+  getIndices: (attr: {
+    config: APMConfig;
+    savedObjectsClient: ISavedObjectsClient;
+  }) => Promise<ApmIndicesConfig>;
+  refetch: (attr: {
+    config: APMConfig;
+    savedObjectsClient: ISavedObjectsClient;
+  }) => Promise<ApmIndicesConfig>;
+  reset: (attr: {
+    config: APMConfig;
+    savedObjectsClient: ISavedObjectsClient;
+  }) => void;
+  resetScheduled: NodeJS.Timeout | null;
+}
+
+export const apmIndicesSettings: ApmIndicesSettings = {
+  getIndices: async ({
+    config,
+    savedObjectsClient,
+  }: {
+    config: APMConfig;
+    savedObjectsClient: ISavedObjectsClient;
+  }) => {
+    if (apmIndicesSettings.indices) {
+      // non blocking call
+      apmIndicesSettings.reset({ config, savedObjectsClient });
+
+      return apmIndicesSettings.indices;
+    }
```

Comment:
I'm not following this: if it's already in the cache, it should be return from cache instead of being refetched, but reset. 

Why reset it here?

Reply:
Caching is always hard. I'd prefer to avoid it. If we really need it, why not use [`lru`](https://github.com/isaacs/node-lru-cache) which is already used in kibana?

It has a `maxAge` that could work here.

Reply:
Btw. is this a shared cache between users? If user A has access to apm data and views this, won't this be available to any other users even if they don't have access?